src/bls12_381/ec.rs

```diff
@@ -166,6 +143,39 @@ macro_rules! curve_impl {
             fn into_projective(&self) -> $projective {
                 (*self).into()
             }
+
+            fn hash(t: Self::Base) -> Self
```

Comment:
I've not checked that the algorithm matches the paper; will do that in another round of review.